node/src/chain/acceptor.rs

```diff
@@ -435,39 +471,60 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
     }
 
     fn selective_update(
-        blk: &Block,
-        txs: &[SpentTransaction],
-        vm: &tokio::sync::RwLockWriteGuard<'_, VM>,
+        block_height: u64,
+        stake_events: &[ContractEvent],
         provisioners_list: &mut tokio::sync::RwLockWriteGuard<
             '_,
             ContextProvisioners,
         >,
     ) -> Result<()> {
         let src = "selective";
-        let changed_prov = Self::changed_provisioners(blk, txs)?;
+        let changed_prov: Vec<_> = stake_events
+            .iter()
+            .filter_map(ProvisionerChange::from_event)
+            .collect();
         if changed_prov.is_empty() {
             provisioners_list.remove_previous();
         } else {
             let mut new_prov = provisioners_list.current().clone();
             for change in changed_prov {
-                let is_stake = change.is_stake();
-                info!(event = "provisioner_update", src, ?change);
-                let pk = change.into_public_key();
-                let prov = pk.to_bs58();
-                match vm.get_provisioner(pk.inner())? {
-                    Some(stake) => {
-                        debug!(event = "new_stake", src, prov, ?stake);
-                        let replaced = new_prov.replace_stake(pk, stake);
-                        if replaced.is_none() && !is_stake {
-                            anyhow::bail!("Replaced a not existing stake")
-                        };
-                        debug!(event = "old_stake", src, prov, ?replaced);
+                let account = change.to_public_key();
+                let value = change.value();
+                info!(
+                    event = "provisioner_update",
+                    src,
+                    account = account.to_bs58(),
+                    value
+                );
+                match &change {
+                    ProvisionerChange::Stake(stake_event) => {
+                        let amount =
+                            StakeAmount::new(stake_event.value, block_height);
+                        let replaced = new_prov.replace_stake(
+                            account,
+                            Stake::new(amount.value, amount.eligibility),
+                        );
+                        if let Some(replaced) = replaced {
+                            if replaced.value() > MINIMUM_STAKE {
```

Comment:
shouldn't this check `value>0`? an existing stake could go lower than MINIMUM_STAKE due to slashing